src/beamlime/executables/prototypes.py

```diff
@@ -0,0 +1,124 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+import argparse
+import pathlib
+from contextlib import contextmanager
+from typing import Any, Optional
+
+from beamlime import Factory
+from beamlime.applications.daemons import Prototype
+from beamlime.logging import BeamlimeLogger
+
+
+def mini_prototype_factory() -> Factory:
+    return Factory(Prototype.collect_default_providers())
+
+
+@contextmanager
+def temporary_factory(
+    prototype_factory: Factory,
+    parameters: Optional[dict[type, Any]] = None,
+    providers: Optional[dict[type, Any]] = None,
+):
+    from beamlime.constructors import (
+        multiple_constant_providers,
+        multiple_temporary_providers,
+    )
+
+    tmp_factory = Factory(prototype_factory.providers)
+    with multiple_constant_providers(tmp_factory, parameters):
+        with multiple_temporary_providers(tmp_factory, providers):
+            yield tmp_factory
```

Comment:
Why do we need this?

Reply:
It was to build application instances easily from a factory without affecting existing one.